consensus/service/src/byzantine_ledger/mod.rs

```diff
@@ -311,6 +315,25 @@ impl Drop for ByzantineLedger {
     }
 }
 
+fn make_metadata_provider(
+    quorum_set: QuorumSet,
+    enclave: impl ReportableEnclave + Send + Sync + 'static,
+    msg_signing_key: Arc<Ed25519Pair>,
+) -> BlockMetadataProvider {
+    Arc::new(move |block_data| {
+        let verification_report = enclave.get_ias_report().expect("failed to get AVR");
+        let contents = BlockMetadataContents::new(
+            block_data.block().id.clone(),
+            quorum_set.clone(),
+            verification_report,
+        );
+        Some(
+            BlockMetadata::from_contents_and_keypair(contents, &msg_signing_key)
+                .expect("failed to sign metadata"),
+        )
+    })
+}
+
```

Comment:
If this is meant to be adding a consensus node's signature to blocks it synced from other nodes - this is currently NOT where it should be. 

Ledger sync currently ONLY validates that the responders are in a quorum set, but not that they've come to consensus on a value. It simply selects the value from the first node in the set: https://github.com/mobilecoinfoundation/mobilecoin/blob/master/ledger/sync/src/ledger_sync/ledger_sync_service.rs#L148. 

From my understanding it, if it's had to catch up it doesn't then tell the other nodes what value its selected, it just fast forwards to new blocks, which wouldn't then set off SCP detection of ill-behavior from old blocks.

If an actor is participating in a valid sense with SCP, but participating maliciously in the network, it could easily feed bad blocks this way and then the consensus node would be signing those bad blocks. This is could at worst fork the network but more likely stop it and require database mitigation. 

From an SCP perspective, previous blocks could be considered "externalized SCP slots" - but it absolutely need to go through full SCP validation of quorum intersection before we sign it in this way.